WMAgent fails to inject some files into rucio

**Impact of the bug**
Bug causes DBS-rucio inconsistency.

**Describe the bug**
Unified [detected](https://cms-unified.web.cern.ch/cms-unified/showlog/?search=task_SUS-RunIIFall17NanoAODv7-00340) a mismatch between DBS and Rucio (well, rucio/phedex) in the number of files for a NanoAOD output dataset. Indeed, if one compares the DBS block [file listing](https://cmsweb.cern.ch/das/request?instance=prod/global&input=file+block%3D%2FSMS-T1tttt_TuneCP2_13TeV-madgraphMLM-pythia8%2FRunIIFall17NanoAODv7-PUFall17Fast_Nano02Apr2020_pilot_102X_mc2017_realistic_v8_ext1-v1%2FNANOAODSIM%2377be8101-f733-4a03-b8df-a6bd3a2e1d8b) with
```
rucio list-content cms:/SMS-T1tttt_TuneCP2_13TeV-madgraphMLM-pythia8/RunIIFall17NanoAODv7-PUFall17Fast_Nano02Apr2020_pilot_102X_mc2017_realistic_v8_ext1-v1/NANOAODSIM#77be8101-f733-4a03-b8df-a6bd3a2e1d8b
```
we see that the following two files are in the block in DBS but not in rucio:
```
/store/mc/RunIIFall17NanoAODv7/SMS-T1tttt_TuneCP2_13TeV-madgraphMLM-pythia8/NANOAODSIM/PUFall17Fast_Nano02Apr2020_pilot_102X_mc2017_realistic_v8_ext1-v1/10000/5CDFB6D9-D1B1-B242-9504-F5547B3DE763.root
/store/mc/RunIIFall17NanoAODv7/SMS-T1tttt_TuneCP2_13TeV-madgraphMLM-pythia8/NANOAODSIM/PUFall17Fast_Nano02Apr2020_pilot_102X_mc2017_realistic_v8_ext1-v1/10000/31AA4395-4218-334E-AEF9-96082A6867D8.root
```
Both files are present on storage at `T2_US_Wisconsin`, the origin site for the block.

**How to reproduce it**
Unknown

**Expected behavior**
All files in DBS should be part of the block in rucio.

Here are some few files that are valid in dbs and in Rucio, but they don't have physical replica yet:

`/store/mc/RunIIFall17NanoAODv7/Wprimetotb_M4400W880_RH_TuneCP5_13TeV-madgraph-pythia8/NANOAODSIM/PU2017_12Apr2018_Nano02Apr2020_102X_mc2017_realistic_v8-v1/10000/EBDEA156-F0BC-9144-AD53-F516F74F0A24.root
/store/data/Run2016G/ZeroBias/NANOAOD/Nano02Dec2019-v1/20000/7559D271-9426-4342-AC5D-1823C8AA6129.root
/store/data/Run2016G/ZeroBias/NANOAOD/Nano02Dec2019-v1/240000/D4B3F955-6768-C747-B536-46D93A2BBB85.root
/store/data/Run2016G/ZeroBias/NANOAOD/Nano02Dec2019-v1/240000/42526BA7-A04C-C246-B69B-C85D61B4C0F6.root`


`$ xrdfs cms-xrd-global.cern.ch locate -d -m /store/mc/RunIIFall17NanoAODv7/Wprimetotb_M4400W880_RH_TuneCP5_13TeV-madgraph-pythia8/NANOAODSIM/PU2017_12Apr2018_Nano02Apr2020_102X_mc2017_realistic_v8-v1/10000/EBDEA156-F0BC-9144-AD53-F516F74F0A24.root
[ERROR] Server responded with an error: [3011] No valid location found

$ xrdfs cms-xrd-global.cern.ch locate -d -m  /store/data/Run2016G/ZeroBias/NANOAOD/Nano02Dec2019-v1/20000/7559D271-9426-4342-AC5D-1823C8AA6129.root

[ERROR] Server responded with an error: [3011] No valid location found

$ xrdfs cms-xrd-global.cern.ch locate -d -m  /store/data/Run2016G/ZeroBias/NANOAOD/Nano02Dec2019-v1/240000/D4B3F955-6768-C747-B536-46D93A2BBB85.root

[ERROR] Server responded with an error: [3011] No valid location found

$ xrdfs cms-xrd-global.cern.ch locate -d -m  /store/data/Run2016G/ZeroBias/NANOAOD/Nano02Dec2019-v1/240000/42526BA7-A04C-C246-B69B-C85D61B4C0F6.root

[ERROR] Server responded with an error: [3011] No valid location found`

--- a/src/python/Utils/FileTools.py
+++ b/src/python/Utils/FileTools.py
@@ -49,7 +49,7 @@ def calculateChecksums(filename):
     if len(cksumStdout) != 2 or int(cksumStdout[1]) != filesize:
         raise RuntimeError("Something went wrong with the cksum calculation !")
 
-    return ("%x" % (adler32Checksum & 0xffffffff), "%s" % cksumStdout[0])
+    return (format(adler32Checksum & 0xffffffff, '08x'), "%s" % cksumStdout[0])
 
 
 def tail(filename, nLines=20):

--- a/test/python/Utils_t/FileTools_t.py
+++ b/test/python/Utils_t/FileTools_t.py
@@ -68,9 +68,8 @@ class testFileTools(unittest.TestCase):
         silly = "This is a rather ridiculous string"
         filename = os.path.join(self.testDir, 'fileInfo.test')
 
-        f = open(filename, 'w')
-        f.write(silly)
-        f.close()
+        with open(filename, 'w') as fObj:
+            fObj.write(silly)
 
         info = FileTools.getFileInfo(filename=filename)
         self.assertEqual(info['Name'], filename)
@@ -83,6 +82,27 @@ class testFileTools(unittest.TestCase):
         self.assertIsNotNone(fullPath)
         fullPath = FileTools.getFullPath("this_shouldnt_be")
         self.assertEqual(fullPath, None)
+
+    def testChecksum(self):
+        """
+        Test file checksums calculation
+        """
+        filename = os.path.join(self.testDir, 'fileInfo.test')
+        with open(filename, 'w') as fObj:
+            fObj.write("")
+
+        (adler32, cksum) = FileTools.calculateChecksums(filename=filename)
+        self.assertEqual(adler32, "00000001")
+        self.assertEqual(cksum, "4294967295")
+
+        silly = "This is a rather ridiculous string"
+        with open(filename, 'w') as fObj:
+            fObj.write(silly * 100001)
+        (adler32, cksum) = FileTools.calculateChecksums(filename=filename)
+        self.assertEqual(adler32, "827db5b1")
+        self.assertEqual(cksum, "3774692924")
+        return
+
 
 if __name__ == "__main__":
     unittest.main()

